### ISSUE TITLE
'React' module used in JSX syntax is not considered

### ISSUE THREAD

**Issue by rhysd** [eslint/eslint#4131]
Test code is following.

```javascript
import React from 'react';
import ReactDOM from 'react-dom';
ReactDOM.render(
        <div>Hello</div>,
        document.querySelector('.app')
    );
```

With above code, eslint reports that `React` is not used.

```
/home/rhysd/Dev/github.com/rhysd/Tilectron/renderer/index.jsx
  20:8  error  "React" is defined but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

However, `React` module is actually used in `<div>` component.  When I removed the import statement, I got below runtime error in DevTools on Chrome.

```
Uncaught ReferenceError: React is not defined
```

I think eslint should consider the modules used in JSX syntax.

My eslint config is [here](https://gist.github.com/rhysd/22952d8688145adb093f)

**Comment by platinumazure** [comment]
Are you using [eslint-plugin-react](https://www.npmjs.com/package/eslint-plugin-react)? If not, you probably should. ESLint will parse JSX but will not apply React semantics.

**Comment by hzoo** [comment]
Looking at the config, @rhysd is using it but doesn't have any rules on - try ` "react/jsx-uses-react": 1`

**Comment by rhysd** [comment]
@platinumazure 

Thank you for the advice.  I'll check `eslint-plugin-react` document.

**Comment by rhysd** [comment]
I read [the document](https://www.npmjs.com/package/eslint-plugin-react). I understood that rules for React.js are not applied by default.  I could resolve this issue.  Thank you @platinumazure & @hzoo!
